docs/extending_tilt_with_bazel_support.md

```diff
@@ -0,0 +1,289 @@
+kind: Deployment
+metadata:
+  labels: {app: snack, owner: varowner}
+  name: varowner-snack
+spec:
+  selector:
+    matchLabels: {app: snack, owner: varowner}
+  template:
+    metadata:
+      labels: {app: snack, owner: varowner, tier: web}
+    spec:
+      containers:
+      - image: bazel/snack@sha256:5a7e2fd3bd39a841cad0f009d605e7f5337a50d2cf641c50ebc03148d8257a0b
+        name: snack
+        ports:
+        - {containerPort: 8083}
+        resources:
+          requests: {cpu: 10m}
+```
+
+With just this we have enough to write our Tiltfile. We'll be using the `local` function to call Bazel and `k8s_yaml` to tell Tilt about the YAML that Bazel generates. Let's write a function that given a Bazel target name returns the YAML for that target.
+
+```python
+def bazel_k8s(target):
+  return local("bazel run %s" % target)
+```
+
+Easy enough, now in the main body of our Tiltfile we can call it like so:
+
+```python
+k8s_yaml(bazel_k8s(":snack-server"))
+```
+
+And if we Tilt up we should see that Tilt calls Bazel to build the YAML, and by extension its image and then Tilt takes that YAML and deploys it to the cluster. We get all the awesomeness of the HUD with three lines of code in a Tiltfile.
+
+### Building Images
+Now for the second component: images. Let's follow the same strategy and figure out the Bazel command that will just build an image and then integrate it in to the Tiltfile.
+
+From reading about rules_k8s I've learned that it uses rules_docker under the hood for building images. Another [quick glance at rules_docker's documentation](https://github.com/bazelbuild/rules_docker#using-with-docker-locally) reveals that `bazel run //:targetname` will load the image in to the configured docker daemon. Perfect, that's all we need!
+
+To integrate this functionality in to the Tiltfile we will make use of just one function: `custom_build`. Custom build takes a command that produces a docker image and tells Tilt about it so it can track it and manage its insertion in to the YAML that we provided previously. Here's what it looks like:
+
+```python
+custom_build(
+  'bazel/snack',
+  'bazel run //snack:image',
+  [],
+  tag="image",
+)
+```
+
+This tells Tilt to run the command (`bazel run //snack:image`) and to assert after it has completed that an image named bazel/snack will exist with the tag "image". If it doesn't then the build fails.
+
+We know what the image will be ahead of time because of Bazel's deterministic nature. Bazel uses a formula for determining the name of the image that it will push: "bazel/" + the directory that the build file is in". This image is tagged with the name of the rule that created it, which in our case is "image".
+
+The third argument, the empty list, is a list of dependencies. We'll fill that in later. For now let's wrap this in a function called `bazel_build` and add it to the main section of our Tiltfile
+
+```python
+k8s_yaml(bazel_k8s(":snack-server"))
+bazel_image('bazel/snack', '//snack:image')
```

Comment:
where does the bazel_image function come from?

Reply:
whoops, this is supposed to be `bazel_k8s`.